### ISSUE TITLE
Impl `GeozeroGeometry` for concrate type should impls srid() and dims()

### ISSUE THREAD

**Issue by ariesdevil** [georust/geozero#195]
Currently only `geos::Geometry<'_>` impl all funcs in `GeozeroGeometry`, other types like `Ewkb` will use the default impl in trait definition.

The default impls are as follows so we cannot get `dims` and `srid` for types like `Ewkb`:
```rust
    fn dims(&self) -> CoordDimensions {
        CoordDimensions::xy()
    }
    /// SRID of geometry
    fn srid(&self) -> Option<i32> {
        None
    }
```

**Comment by ariesdevil** [comment]
And I need parse srid manually when do transform: https://github.com/datafuselabs/databend/pull/14615/files#diff-9be506d723a6d34b70933330d0e09aac81d43cb005f941978b94acad13cdb622R154

**Comment by netthier** [comment]
Just noticed this myself, I'd expect the `srid()` method on an `Ewkb` to return the encoded SRID.

**Comment by pka** [comment]
I agree that `Ewkb` and others should implement `srid()` and `dims()`. PR welcome!
